widgets/matplotlib/simulation/element.py

```diff
@@ -100,46 +107,51 @@ def __init__(self, parent, element: Element, parent_tab,
 
         # Circle for showing the recoil color
         self.circle = Circle(color)
+        instance_buttons.append(self.circle)
 
+        change_recoil_element_info = QtWidgets.QPushButton()
         draw_spectrum_button = QtWidgets.QPushButton()
-        draw_spectrum_button.setIcon(icon_manager.get_icon(
-            "energy_spectrum_icon.svg"))
-        draw_spectrum_button.setSizePolicy(QtWidgets.QSizePolicy.Fixed,
-                                           QtWidgets.QSizePolicy.Fixed)
-
-        draw_spectrum_button.clicked.connect(lambda: self.plot_spectrum(
-            spectra_changed=spectra_changed))
-        draw_spectrum_button.setToolTip("Draw energy spectra")
-
         settings_button = QtWidgets.QPushButton()
-        settings_button.setIcon(icon_manager.get_icon("gear.svg"))
-        settings_button.setSizePolicy(QtWidgets.QSizePolicy.Fixed,
-                                      QtWidgets.QSizePolicy.Fixed)
-        settings_button.clicked.connect(
-            lambda: self.open_element_simulation_settings(
-                settings_updated=settings_updated))
-        settings_button.setToolTip("Edit element simulation settings")
-
         add_recoil_button = QtWidgets.QPushButton()
-        add_recoil_button.setIcon(icon_manager.get_icon("edit_add.svg"))
-        add_recoil_button.setSizePolicy(QtWidgets.QSizePolicy.Fixed,
-                                        QtWidgets.QSizePolicy.Fixed)
-        add_recoil_button.clicked.connect(lambda: self.add_new_recoil(
-            spectra_changed=spectra_changed,
-            recoil_name_changed=recoil_name_changed))
-        add_recoil_button.setToolTip("Add a new recoil to element")
+
+        buttons.append(change_recoil_element_info)
+        buttons.append(draw_spectrum_button)
+        buttons.append(settings_button)
+        buttons.append(add_recoil_button)
+
+        icons = ["measuring_unit_settings.svg",
+                 "energy_spectrum_icon.svg",
+                 "gear.svg", "edit_add.svg"]
+        tool_tips = ["Change the color of the element",
+                     "Draw energy spectra",
+                     "Edit element simulation settings",
+                     "Add a new recoil to element"]
+        method_calls = [parent.change_recoil_element_info,
+                        lambda: self.plot_spectrum(
+                            spectra_changed=spectra_changed),
+                        lambda: self.open_element_simulation_settings(
+                            settings_updated=settings_updated),
+                        lambda: self.add_new_recoil(
+                            spectra_changed=spectra_changed,
+                            recoil_name_changed=recoil_name_changed)
+                        ]
```

Comment:
I'm not sure if iterating GUI elements like this is a good thing. Consitency is important, and no other part of the code does this (I think). While the previous style is repetitive, it is more readable and straightforward.

---

Spreading individual items over multiple lists is error-prone. A safer alternative would be to use a per-item dict like so:

```python
color_settings = {
    "icon": "measuring_unit_settings.svg",
    "tooltip": "Change the color of the element",
    "callback": parent.change_recoil_element_info
}

```

`namedtuple` from `collection` is another option, search the code for examples on that.

Reply:
I agree with that error-prone. But otherwise I think it's a matter of taste. I'll roll that back.

Reply:
Check